src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,102 @@
+package duke.storage;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import duke.enums.ErrorCodes;
+import duke.exceptions.DukeException;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.TaskList;
+import duke.tasks.ToDo;
+
+public class Storage {
+    private String filePath;
+    private ArrayList<Task> taskList;
+
+    /**
+     * Storage constructor.
+     *
+     * @param filePath String path of the duke.txt file.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Load the list of tasks from the file specified.
+     *
+     * @return An ArrayList containing the loaded tasks.
+     * @throws DukeException DukeException thrown if file is not found or any line not being parsed succesfully.
+     */
+    public ArrayList<Task> load() throws DukeException {
+        taskList = new ArrayList<Task>();
+        try {
+            Scanner readFile = new Scanner(new File(this.filePath));
+            int lineNo = 1;
+            while (readFile.hasNextLine()) {
+                loadTask(readFile.nextLine(), lineNo);
+                lineNo++;
+            }
+            readFile.close();
+            return taskList;
+        } catch (IOException e) {
+            throw new DukeException(ErrorCodes.FILE_NOT_FOUND);
+        }
+    }
+
+    /**
+     * Load the individual lines from the input file.
+     *
+     * @param taskString A string of the current line to be loaded from storage.
+     * @param lineNo The current index of the line in the file.
+     * @throws DukeException DukeException thrown if any line cannot be loaded.
+     */
+    private void loadTask(String taskString, int lineNo) throws DukeException {
+        String[] splitString = taskString.split(" \\| ");
```

Comment:
Bug: Since | isn't a forbidden character in the task description, a user adding a task with "|" in the description will be unable to load this task from the file.
Suggestion: Either escape "|" in the save data format, or use newlines to delimit task arguments.